### ISSUE TITLE
<w:lastRenderedPageBreak />

### ISSUE THREAD

**Issue by dwasyl** [mwilliamson/python-mammoth#61]
As part of using `mammoth` I've had a need to detect natural page breaks. While Word doesn't mark pages, it does insert `<w:lastRenderedPageBreak />` into the document at the flowed page breaks (seemingly at column breaks as well).

To detect these, I hacked together a system adding a detector to `docx/body_xml.py`'s `def inline(element)` to detect `<w:lastRenderedPageBreak />` elements and call a function that inserts dummy text that I find and replace later.

Realizing that this isn't a very reusable way of looking for these rendered page breaks, I was wondering if you had a preferred way of detecting/inserting these extra tags in the HTML. If so, I'd be happy to try turning it into a proper PR.

**Comment by GitBruno** [comment]
You can use a [stylemap](https://github.com/mwilliamson/python-mammoth#styles):

    br[type='page'] => br.page

See #20

**Comment by mwilliamson** [comment]
The style maps only support explicit breaks, `w:lastRenderedPageBreak` elements are currently ignored by the parser. We could add support for them into the parser -- from my (possibly incorrect) recollection, the slightly tricky thing about them is that they end up in the middle of text, unlike explicit breaks, so need handling differently.

**Comment by dwasyl** [comment]
Definitely, currently I hacked together a secondary detection that adds a marker to the text for me to pick up later. Even if they are supported through the parser with a special break tag (or user configurable) it would be handy.

They can appear in the middle of text - they appear anywhere a page break has been rendered by Word. That said, detecting them isn't anything special.

The biggest issue with `w:lastRenderedPageBreak` is that they are difficult to rely on as they are only inserted by Word and not visible/configurable in any way.

**Comment by dwasyl** [comment]
@mwilliamson is there a way you'd prefer to hook this in? I could work on a PR, but what I've done is simply have it replace the `w:lastRenderedPageBreak` with some marker text I found later. Realizing this isn't the best overall approach, I could try refactoring it another way?

**Comment by menbatisiunissart** [comment]
Im also interested in this feature @dwasyl would you be able to share what you ended up doing?

**Comment by MadMaxINDIAN** [comment]
@dwasyl Yes! I am also interested in the final result of the mammoth repository modified by you

**Comment by dwasyl** [comment]
@mwilliamson I was just looking to update my fork of `mammoth` that makes use of the `w:lastRenderedPageBreak` detection, and rather than maintain a separate fork would love to just send it back to you as a PR. As you can see from the hack-job I created ([https://github.com/dwasyl/python-mammoth/commit/624d81894ac5d0b5249ecfd4cf74e749763c0d47](https://github.com/dwasyl/python-mammoth/commit/624d81894ac5d0b5249ecfd4cf74e749763c0d47)), I made a very similar modificatino to detect the `w:lastRenderedPageBreak` and insert a keyword I could detect in the next stage of my processing. This likely isn't a reasonable solution for the library, but do you have something in mind that would be reasonable (and would accept)?

I thought of a few different methods including `<hr class="pagebreak">`, but I wasn't sure what would best fit the style of `mammoth`. If you have any guidance, I'd be happy to include it, although it's a pretty simple fix.

**Comment by dwasyl** [comment]
@MadMaxINDIAN @menbatisiunissart apologies for missing your questions, you can see what I did here [https://github.com/dwasyl/python-mammoth/commit/624d81894ac5d0b5249ecfd4cf74e749763c0d47](https://github.com/dwasyl/python-mammoth/commit/624d81894ac5d0b5249ecfd4cf74e749763c0d47), it wasn't anything special but it met the need for my purposes as I was just counting pages in the resulting document. Hope that helps!

**Comment by dwasyl** [comment]
@mwilliamson I've been using this in a fork for ... a lot of years. Any chance I could turn it into a PR or you can add it as a basic feature?

The idea being to add some kind of unique flag when there's a pagebreak. I've always inserted something simple like `renderedpage` however, it's likely more appropriate to insert something like `<!-- rendered pagebreak -->` or some such thing.

It's quite a handy flag to try and map out how word has treated the document.

**Comment by mwilliamson** [comment]
The simplest thing to do would be to add a style mapping to insert an HTML element, for instance:

```
last-rendered-page-break => span.last-rendered-page-break
```

I imagine that wouldn't be too tricky if that would solve your problem?
